In [2]:
import h5py

Let's load the entire file.

In [6]:
f = h5py.File("/data/genome-attention/l131k_w128.h5", 'r')

In [7]:
list(f.keys())

['pool_width', 'target_ids', 'target_labels', 'target_strands']

In [5]:
f['target_labels']

<HDF5 dataset "target_labels": shape (4229,), type "|S64">

**Training In**

Size: 13,047 

Length: 131,072 bp

Shape: 131,072 x 4

In [27]:
f['train_in']

<HDF5 dataset "train_in": shape (13047, 131072, 4), type "|b1">

**Training Out**

Size: 13,047 

Length: 1024 buckets 

Labels: 4229 labels

Shape: 1024 x 4229


In [25]:
f['train_out']

<HDF5 dataset "train_out": shape (13047, 1024, 4229), type "<f2">

In [16]:
f['valid_in']

<HDF5 dataset "valid_in": shape (726, 131072, 4), type "|b1">

In [17]:
f['valid_out']

<HDF5 dataset "valid_out": shape (726, 1024, 4229), type "<f2">

In [19]:
f['test_in']

<HDF5 dataset "test_in": shape (760, 131072, 4), type "|b1">

In [18]:
f['test_out']

<HDF5 dataset "test_out": shape (760, 1024, 4229), type "<f2">